addons/website/static/src/js/website.contentMenu.js

```diff
@@ -236,6 +238,7 @@ var EditMenuDialog = widget.Dialog.extend({
     add_menu: function () {
         var self = this;
         var dialog = new MenuEntryDialog(this, {menu_link_options: true}, undefined, {});
+        dialog.$modal.addClass('o_contentMenu_link_modal');
```

Comment:
Unfortunately not :/
The XML template specify the inner content and, on the JS side, the `Dialog` widget  attribute `dialogClass` is added to the inner `.modal-body`.